sessionrecording/connect.go

```diff
@@ -54,72 +58,221 @@ func ConnectToRecorder(ctx context.Context, recs []netip.AddrPort, dial func(con
 		}
 		attempts = append(attempts, attempt)
 
-		// We dial the recorder and wait for it to send a 100-continue
-		// response before returning from this function. This ensures that
-		// the recorder is ready to accept the recording.
-
-		// got100 is closed when we receive the 100-continue response.
-		got100 := make(chan struct{})
-		ctx = httptrace.WithClientTrace(ctx, &httptrace.ClientTrace{
-			Got100Continue: func() {
-				close(got100)
-			},
-		})
-
-		pr, pw := io.Pipe()
-		req, err := http.NewRequestWithContext(ctx, "POST", fmt.Sprintf("http://%s:%d/record", ap.Addr(), ap.Port()), pr)
+		var pw io.WriteCloser
+		var errChan <-chan error
+		var err error
+		hc := clientHTTP2(dialCtx, dial)
+		if supportsV2(ctx, hc, ap) {
```

Comment:
When V2 is supported, this introduces a bit of latency. I think you could eliminate this by just immediately connecting to the V2 endpoint and if it returns a 404, fall back to V1.

Reply:
For some reason I don't quite understand, hitting the old HTTP/1 recorder using `connectV2` gets stuck on `hc.Do` here https://github.com/tailscale/tailscale/pull/14105/files#diff-2e2652218f2c8dfa2d3ba160dcf15e864273368f4308c8345429337468ce3977R187-R197. Adding debug logs to the old recorder I can see that it checks the path and returns a 404. But the client here doesn't return from `hc.Do` to check `resp.StatusCode`.

But hitting the `HEAD` endpoint with the same http/2 client does respond immediately with error `Head "http://[fd7a:115c:a1e0::7801:9278]:80/v2/record": unexpected EOF`. My assumption is that `HEAD` requests are treated somehow special in http/2 code allowing them to fail early :shrug:

Reply:
How weird! I haven't played much with HTTP/2, let me mess with it for a few minutes to see if anything jumps out at me.

Reply:
So yeah, when doing the HTTP2 `POST` to an HTTP1 endpoint, if, and only if, I close the request body, then do I get the same failure as on the `HEAD` request. Of course that's not an option for us, so I also don't see a way around doing the HEAD request.

Maybe it's worth leaving a comment about why we have to make a `HEAD` request in case this question comes up again?

Reply:
Yeah, it's strange. Maybe it's not HEAD vs POST, but just a request with a body vs nil body.
Added a comment :ok_hand: